worlds/stardew_valley/logic/money_logic.py

```diff
@@ -11,10 +12,14 @@
 from ..data.shop import ShopSource
 from ..options import SpecialOrderLocations
 from ..stardew_rule import StardewRule, True_, HasProgressionPercent, False_, true_
-from ..strings.ap_names.event_names import Event
 from ..strings.currency_names import Currency
 from ..strings.region_names import Region, LogicRegion
 
+if typing.TYPE_CHECKING:

```

Comment:
What's up with that?

Reply:
I assumed it was some circular dependency, but maybe not

Reply:
I'm not a big fan of opening the pandora's box of just... accepting circular dependencies, instead of trying to make the design better and avoid them.

Do you think there would be a way to split things more to avoid it?

Reply:
Maybe... We could split rules based on the options. Building vanilla and building progressive both don't have circular dependencies when taken individually. They is only one circular dependencies when they are both in the same logic file. 

Vanilla: building -> money -> shipping -> true
Progressive: money -> shipping -> building -> ap item

I'm just afraid that splitting logic files by options would be an even bigger pandora's box. It's simple for building because there is only one option involved, but other logic might depend on more options... Fishing for instance depends on special orders and ginger island... Might be simpler to accept it now and reassess when features are properly introduced.

Reply:
Alright, I'll defer to your judgement on it